### PR TITLE
Small improvements for RefMan generation

### PR DESCRIPTION
**Pull request by FabienLelaquais**

### CODE REVIEW

File: src/taipy/core/config/task_config.py [hunk 1]
```diff
@@ -23,20 +23,24 @@
 
 class TaskConfig(Section):
     """
-    Holds all the configuration fields needed to create an actual `Task^` from the `TaskConfig`.
+    Configuration fields needed to create an actual `Task^`.
```
Comment (jrobinAV) [inline src/taipy/core/config/task_config.py]:
```suggestion
    Configuration fields needed to instantiate an actual `Task^`.
```

**Review by jrobinAV — DISMISSED** [review]

**Review by jrobinAV — DISMISSED** [review]

File: src/taipy/core/task/task.py [hunk 2]
```diff
@@ -178,7 +178,7 @@ def submit(
         force: bool = False,
         wait: bool = False,
         timeout: Optional[Union[float, int]] = None,
-    ):
+    ) -> "Job":
```
Comment (jrobinAV) [inline src/taipy/core/task/task.py]: I am not sure we can do that. 
Each Job references a Task. So the Task class cannot reference the Job class without a circular import.

Comment (FabienLelaquais) [reply src/taipy/core/task/task.py]: There is no import at this point...

**Review by jrobinAV — DISMISSED** [review]
Tests should pass now...

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**